lua/lspconfig/jdtls.lua

```diff
@@ -71,44 +82,44 @@ end
 
 configs[server_name] = {
   default_config = {
-    cmd = cmd,
-    cmd_env = {
-      JAR = vim.fn.getenv 'JAR',
-      GRADLE_HOME = vim.fn.getenv 'GRADLE_HOME',
+    cmd = {
+      get_java_executable(),
+      '-Declipse.application=org.eclipse.jdt.ls.core.id1',
+      '-Dosgi.bundles.defaultStartLevel=4',
+      '-Declipse.product=org.eclipse.jdt.ls.core.product',
+      '-Dlog.protocol=true',
+      '-Dlog.level=ALL',
+      '-Xms1g',
+      '-Xmx2G',
+      '-jar', get_jdtls_jar(),
+      '-configuration', get_jdtls_config(),
+      '-data', get_workspace_dir(),
+      '--add-modules=ALL-SYSTEM',
+      '--add-opens java.base/java.util=ALL-UNNAMED',
+      '--add-opens java.base/java.lang=ALL-UNNAMED',
     },
     filetypes = { 'java' },
-    root_dir = function(fname)
-      for _, patterns in ipairs(root_files) do
-        local root = util.root_pattern(unpack(patterns))(fname)
-        if root then
-          return root
-        end
-      end
-      return vim.fn.getcwd()
-    end,
+    root_dir = util.root_pattern({
```

Comment:
People asked for the mult-root fallback, so I'm concerned about changing it.

Reply:
I didn't saw the getcwd and added it back. I don't see why the multi root would be necessary, are there any issues related to that?

Reply:
https://github.com/neovim/nvim-lspconfig/issues/455

Reply:
Also I got asked about this on matrix IIRC and dm'd a few times